src/uct/ib/mlx5/dc/dc_mlx5_ep.h

```diff
@@ -335,6 +335,13 @@ uct_dc_mlx5_is_hw_dci(const uct_dc_mlx5_iface_t *iface, uint8_t dci)
     return dci == iface->tx.hybrid_hw_dci;
 }
 
+static UCS_F_ALWAYS_INLINE int
+uct_dc_mlx5_is_dci_shared_or_hw(const uct_dc_mlx5_iface_t *iface, uint8_t dci)
+{
+    return uct_dc_mlx5_iface_is_dci_shared(iface) ||
+           uct_dc_mlx5_is_hw_dci(iface, dci);
```

Comment:
can we optimize this to be a single branch? it is used in uct_dc_mlx5_iface_dci_get

Reply:
I guess replacing `||` with `+`?

Reply:
both these checks are checking the policy field, maybe we can a bitmap based check or something like "policy >= UCT_DC_TX_POLICY_DCS_HYBRID" ?

Reply:
is_hw_dci just checks `dci == iface->tx.hybrid_hw_dci` so `>= DCS_HYBRID` won't help, we can just use bitwise `|` between is_shared and is_hw_dci if that's somehow better